Free/retract

```diff
@@ -0,0 +1,15 @@
+   let Monad =
+         https://raw.githubusercontent.com/FormationAI/dhall-bhat/8afbcee6dfd092764df9e9456fe2a2650748dce2/Monad/Type
+
+in    λ ( f
+        : Type → Type
+        )
+    → λ(a : Type)
+    → λ(monad : Monad f)
+    → let join =
+            ( https://raw.githubusercontent.com/FormationAI/dhall-bhat/6f66fc588c9d49a662794a32b29962341c373eb4/Monad/package.dhall
+              f
+              monad
+            ).join
+      
+      in  λ(m : ./Type f a) → m (f a) (monad.pure a) (join a)
```

Comment:
```dhall
  λ(f : Type → Type)
→ λ(monad : Monad f) -- note instance immediately after type
→ λ(a : Type)
→ ./recursive.cata
  a
  (  λ(free : ../../FreeF/Type f a (f a))
   → merge { Pure = monad.pure, Free = join } free)
```
and the `merge` would be cleaner if I ever got my changes into the spec, then it could be
```dhall
→ ./recursive.cata (f a) (merge { Pure = monad.pure, Free = join })
```

Reply:
Basically, anything that applies `m` is going to instead do `./recursive.cata … merge`